src/Agent.Listener/Agent.cs

```diff
@@ -174,93 +174,97 @@ private void CtrlCHandler(object sender, EventArgs e)
             IJobDispatcher jobDispatcher = null;
             try
             {
-                bool disableAutoUpdate = !string.IsNullOrEmpty(Environment.GetEnvironmentVariable("agent.disableupdate"));
-                bool autoUpdateInProgress = false;
-                Task<bool> selfUpdateTask = null;
-                jobDispatcher = HostContext.CreateService<IJobDispatcher>();
-                while (!token.IsCancellationRequested)
+                using (var notification = HostContext.GetService<IJobNotification>())
```

Comment:
to clarify for other readers, only this line is new (creating the notification service) and the following line `notifcation.StartClient[...]`